packages/remix/src/client/ClerkApp.tsx

```diff
@@ -1,14 +1,22 @@
-import { useLoaderData } from '@remix-run/react';
+import { useLoaderData, useRouteError } from '@remix-run/react';
 import React from 'react';
 
+import { ClerkErrorBoundary } from '.';
 import type { RemixClerkProviderProps } from './RemixClerkProvider';
 import { ClerkProvider } from './RemixClerkProvider';
 
 type ClerkAppOptions = Partial<Omit<RemixClerkProviderProps, 'navigate' | 'children' | 'clerkState'>>;
 
 export function ClerkApp(App: () => JSX.Element, opts: ClerkAppOptions = {}) {
   return () => {
+    // @ts-expect-error - we know clerkState is in the loader data
     const { clerkState } = useLoaderData();
+    const error = useRouteError();
+
+    if (error) {
+      return ClerkErrorBoundary();
```

Comment:
Don't we need to check the type of error here?
What happens if a loader throws a user-defined error? I think this would trigger the interstitial for v4, isn't that the case?

Reply:
Good catch, yeah I want to test that to see what happens that's why I marked it as draft so I can test it tomorrow and apply a fix.